antarest/study/dao/api/xpansion_dao.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2025, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from antarest.study.business.model.xpansion_model import XpansionCandidate
+
+
+class ReadOnlyXpansionDao(ABC):
+    @abstractmethod
+    def get_all_xpansion_candidates(self) -> list[XpansionCandidate]:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get_xpansion_candidate(self, candidate_id: str) -> XpansionCandidate:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def checks_xpansion_candidate_coherence(self, candidate: XpansionCandidate) -> None:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def checks_xpansion_candidate_can_be_deleted(self, candidate_name: str) -> None:
```

Comment:
It seems to me that we could just have the delete method which would take a name input, and do the check there ?
It would avoid to to a check/get/delete in the remove command